frontend/src/app/workflows/workflows.component.ts

```diff
@@ -112,19 +109,20 @@ export class WorkflowsComponent implements OnInit {
     );
   }
 
-  setIndexFromIframe(index: number, diagramType: 'pages' | 'objects' | 'class') {
+  @HostListener('window:message', ['$event'])
+  handleGlobalMessages(event: MessageEvent) {
+    const messageData = JSON.parse(event.data);
     // It needs to be run in the NgZone because only then angular change detection gets a grip on the change
     this.zone.run(() => {
```

Comment:
HostListener comment applies here too. Also, shouldn't there be a check for the messageData.type? Keep in mind that this event could contain any data, even unrelated to fulibWorkflows.

Reply:
That's what that was. Now it only allows the correct types for fulibWorkflows in that method.